src/core/storage/fileio/s3_api.cpp

```diff
@@ -147,6 +147,20 @@ S3Client init_aws_sdk_with_turi_env(s3url& parsed_url) {
 
   // TODO: add proxy support
   // clientConfiguration.proxyHost = proxy.c_str();
+  if (!get_alternative_ssl_cert_file().empty()) {
+    auto fstaus = get_file_status(get_alternative_ssl_cert_file());
```

Comment:
I don't think this is the right way. I think you should be getting the `clientConfiguration.caPath` value from `get_alternative_ssl_cert_dir()` not from `get_alternative_ssl_cert_file()`.

Also I'm not sure it makes sense but I think the should be possible for both `clientConfiguration.caPath` and `clientConfiguration.caFile` to be set.